token.go

```diff
@@ -16,6 +16,7 @@ const (
 	AT             = "@"
 	EQUAL          = "="
 	PLUS           = "+"
+	COMMA          = ","
```

Comment:
Just thinking out loud for the syntax, these are some other options we have.

One program per require (a bit verbose):
```
Require foo
Require bar
Require baz
```

Whitespace separated (a bit tricky)
```
Require foo bar baz
```

Colon-separated (resembles `$PATH`)
```
Require foo:bar:baz
```

Comma separated (current impl):
```
Require foo,bar,baz
```

Reply:
The comma separated is the most unambiguous, personally.

Reply:
I would prefer the one require per line, actually, otherwise if you have a lot of requirements the line width would get too big.

also, one per line make code reviews easier, as its either adding or remove one line, instead of changing a big one.

Reply:
One require per line makes sense. They are like `import`s in other programming languages.

Just pitching another one:
```sh
Require foo

# or

Require (
    foo
    bar
    neofetch
    fdfind
    btop
)
```
like Golang `import`s.

Granted, it'll be tougher to parse.

Reply:
@abhijit-hota I think we can start with one per line and each needs a require since that more closely matches how the rest of the language works i.e. more command based.

Also, if people really want to keep everything on one line they can do:

```
Require foo Require bar Require baz
```

So, I think this is a decent solution!

Reply:
Makes sense. Will implement this!

Reply:
On second thought, if we implement this syntax, we cannot check for all binaries that are missing altogether. 

As we have to check the `Require` one by one.

Reply:
One way around this is to make `ExecuteRequire` just add the programs to a list on VHS and then when you hit a non-require command evaluate all the requires in that group and clear the list.

Reply:
So you basically group requires:

```
# Will exit early if foo, bar, baz are missing
Require foo
Require bar
Require baz

Type "foo bar baz"
Enter

# Will exit if qux is missing
Require qux
Type "qux"
```

Reply:
Another idea is to do a pre-run of all the commands and filter all the require ones, add them to a list and exit if any are missing.

Reply:
So to summarise, we have the following 2 behaviours we need to decide on. Please correct me if I'm wrong or missing anything.

### Behaviour 1
- All the `Require`s are checked before the actual recording is started.
- Outputs all the missing binaries altogether, if any.
- Note on semantics:
    - If we intend the command to be used anywhere in the code, we have to pre-parse the whole tape. 
    - Another alternative is to an only allow it at the start of a `tape`. 

### Behaviour 2
- Each `Require` is checked individually.
- Outputs only the binary at which it failed.
- Can be used anywhere.

Reply:
Yep! I'm actually okay with either behaviour. Also happy to set a requirement that they are at the top of the file if they want to be used, we have a similar requirement of Output and Set commands.

Reply:
Hmm, I believe its only settings which must be at the top then, if there is a setting in the middle of a tape it will be ignored

Reply:
Imo we can allow requires to be anywhere and exit at the point they are which is a more expected behaviour

Reply:
The following piece of code only checks for Set commands and that's the reason Output isn't ignored wherever is it.

https://github.com/charmbracelet/vhs/blob/1a2716c7684d409ed575a9fd5932c3c2086d711a/evaluator.go#L92-L98

Reply:
@maaslalani The problem I think with executing `Require`s individually is this:

```sh
Require foo
Type foo
Enter

Sleep 15s

Require bar
Type bar
Enter
```

Now the user will not know about the `bar` dependency for like ≈20s. The original issue was to "fail-first" and I think we should do that as soon as possible.

Reply:
I will implement the pre-parsing method and get back to you.

Reply:
Cool, thanks! Yeah I believe we should pre-parse and always encourage (in our examples) to put all the `Require` statements at the top of the file as a best practice.

Reply:
Just implemented it. Take a look. @maaslalani